constants/constants.hpp

```diff
@@ -0,0 +1,235 @@
+// ======================================================================
+// © 2021. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract
+// 89233218CNA000001 for Los Alamos National Laboratory (LANL), which
+// is operated by Triad National Security, LLC for the U.S.
+// Department of Energy/National Nuclear Security Administration. All
+// rights in the program are reserved by Triad National Security, LLC,
+// and the U.S. Department of Energy/National Nuclear Security
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+// ======================================================================
+
+#ifndef CONSTANTS_CONSTANTS_
+#define CONSTANTS_CONSTANTS_
```

Comment:
maybe this include guard should be something like `SINGULARITY_CONSTANTS_CONSTANTS_` or something just in case this guard signature is used elsewhere, which I can imagine happening.

Reply:
nvcc yells at me for `exit`. So I don't think we should add it.